dockerfile_parse/parser.py

```diff
@@ -100,6 +111,26 @@ def __init__(self, path=None, cache_content=False, env_replace=True, parent_env=
         else:
             self.parent_env = {}
 
+    @contextmanager
+    def _open_dockerfile(self, mode):
+        if self.fileobj is not None:
+            try:
+                self.fileobj.seek(0)
+            except IOError:
```

Comment:
Not rewinding the file will cause unexpected behavior when using this class.
Can the "requirement" be that the file must be seekable?

Perhaps this can be enforced with a `seek(0)` call in `__init__` to fail immediately and provide a meaningful error message?